pipeline/data/dataset_importer.py

```diff
@@ -0,0 +1,166 @@
+    "aug-typos": TypoModifier(MIX_PROB),
+    "aug-title": TitleCaseModifier(MIX_PROB),
+    "aug-title-strict": TitleCaseModifier(1.0),
+    "aug-upper": UpperCaseModifier(MIX_PROB),
+    "aug-upper-strict": UpperCaseModifier(1.0),
+    "aug-mix": CompositeModifier(
+        [
+            # TODO: enable typos, issue https://github.com/hplt-project/OpusTrainer/issues/40
+            # TypoModifier(NOISE_RATE),
+            TitleCaseModifier(MIX_PROB),
+            UpperCaseModifier(MIX_PROB),
+        ]
+    ),
+}
+
+
+def run_cmd(cmd: List[str]):
+    result = None
+    try:
+        result = subprocess.run(
+            cmd,
+            universal_newlines=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+        result.check_returncode()
+    except:
+        if result:
+            print(result.stdout)
+        raise
+
+    print(result.stdout)
+
+
+# we plan to use it only for small evaluation datasets
+def augment(output_prefix: str, aug_modifer: str):
+    if aug_modifer not in modifier_map:
+        raise ValueError(f"Invalid modifier {aug_modifer}. Allowed values: {modifier_map.keys()}")
+
+    modifer = modifier_map[aug_modifer]
+
+    # these envs are standard across the pipeline
+    src = os.environ["SRC"]
+    trg = os.environ["TRG"]
+    comp_cmd = os.getenv("COMPRESSION_CMD", "pigz")
+    comp_ext = os.getenv("ARTIFACT_EXT", "gz")
+
+    uncompressed_src = f"{output_prefix}.{src}"
+    uncompressed_trg = f"{output_prefix}.{trg}"
+    compressed_src = f"{output_prefix}.{src}.{comp_ext}"
+    compressed_trg = f"{output_prefix}.{trg}.{comp_ext}"
+
+    if os.path.isfile(uncompressed_src):
+        os.remove(uncompressed_src)
+    if os.path.isfile(uncompressed_trg):
+        os.remove(uncompressed_trg)
+    run_cmd([comp_cmd, "-d", compressed_src])
+    run_cmd([comp_cmd, "-d", compressed_trg])
+
+    with open(uncompressed_src) as f:
+        corpus_src = [line.rstrip("\n") for line in f]
+    with open(uncompressed_trg) as f:
+        corpus_trg = [line.rstrip("\n") for line in f]
+
+    corpus = [f"{src_sent}\t{trg_sent}" for src_sent, trg_sent in zip(corpus_src, corpus_trg)]
+    modified = list(modifer(corpus))
+    modified_src = "\n".join([line.split("\t")[0] for line in modified]) + "\n"
+    modified_trg = "\n".join([line.split("\t")[1] for line in modified]) + "\n"
+
+    with open(uncompressed_src, "w") as f:
+        f.write(modified_src)
+    with open(uncompressed_trg, "w") as f:
+        f.writelines(modified_trg)
+
+    os.remove(compressed_src)
+    os.remove(compressed_trg)
+    run_cmd([comp_cmd, uncompressed_src])
+    run_cmd([comp_cmd, uncompressed_trg])
+    os.remove(uncompressed_src)
+    os.remove(uncompressed_trg)
+
+
+def run_import(type: str, dataset: str, output_prefix: str):
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+
+    if type == "corpus":
+        match = re.search(r"^(\w*)_(aug[a-z\-]*)?_?(.+)$", dataset)
```

Comment:
Suggestion: Please give examples of how this identifier match is supposed to work. It's not clear from this identifier how it's supposed to work or what is valid. This also seems like it would be worth putting it into a utility if this is shared anywhere else in the code. Regexes like this can be a source of bugs, and it's not clear what the exact intent is for this one.

Reply:
added examples